modules/ejbca-ejb/src/org/ejbca/core/ejb/ca/caadmin/CAAdminSessionBean.java

```diff
@@ -2356,6 +2370,12 @@ private void renewCAInternal(final AuthenticationToken authenticationToken, int
                 ca.createOrRemoveLinkCertificate(cryptoToken, createLinkCertificate, certprofile, cceConfig, oldCaCertificate);
                 caSession.editCA(authenticationToken, ca, true);
             }
+            
+            // Put the Signature Algorthm settings back if we changed them to issue the Link certificate.
+            if (bChangedSigAlg) {
```

Comment:
Not needed, I believe, if only using a cloned cert profile as argument to the link cert creation methods?

Reply:
Possible over-kill. I was being careful here, especially with the Token settings. The Cert Profile should not matter if it isn't being saved in the DB.